### ISSUE TITLE
The licensing between the website and the book doesn't seem to match

### ISSUE THREAD

**Issue by duchainer** [SquareBracketAssociates/LearningOOPWithPharo#57]
Hi, the book is great and answered many of my question getting into Samlltalk in general.

I just realized the license [mentioned on the website](https://books.pharo.org/learning-oop/):
```
The content of this book is released under a Creative Commons Attribution-NoCommercial-ShareAlike license.
```
is not the same as the one in the book:
![image](https://user-images.githubusercontent.com/11084784/122829183-ad65ac80-d2b4-11eb-8d93-baae0828dae1.png)

Thanks in advance for your time.

I personally would like to publicly adapt the TDD parts for [glamorous toolkit](https://gtoolkit.com/)'s [Example Driven Development](http://scg.unibe.ch/archive/phd/gaelli-phd.pdf), but if it is prohibiting derivatives like the book license seems to say, I would respect it and would have to keep it for my personal use only.

**Comment by cdlm** [comment]
The BY-NC-ND PDF is quite old, and if you compile from sources now it should say BY-SA, as far as I can tell.
I don't understand why it was licensed differently from all the other books in the first place.

**Comment by duchainer** [comment]
> The BY-NC-ND PDF is quite old, and if you compile from sources now it should say BY-SA, as far as I can tell.
> I don't understand why it was licensed differently from all the other books in the first place.

Yes, indeed, no mention of BY-NC-ND in the latest version of the code, but BY-SA in its place. 
So the issue is rather that the pdf export on the site is a bit old. Thanks for the heads up.

**Comment by Ducasse** [comment]
Hello I will check because I was fed up to write book that other can use and extend without my approval.

**Comment by Ducasse** [comment]
BTW for the example development note that in Pharo with one class and one method you can get the same as with the 100 classes of GT. I think that this is even documented in the Test booklet.

**Comment by Ducasse** [comment]
Thanks for the glitch. I removed the license from the website (it was a copy paste).

**Comment by duchainer** [comment]
> Hello I will check because I was fed up to write book that other can use and extend without my approval.
> I got fed up to give all my time basically for free (this is ok) but in addition to see that people can reuse my work for things I do not want/like. This is why I set this license.

Really sorry for those past abuses. I won't add any more mediocre derivatives to worry about. I'll keep my notes to myself. ^^"

**Comment by duchainer** [comment]
> BTW for the example development note that in Pharo with one class and one method you can get the same as with the 100 classes of GT. I think that this is even documented in the Test booklet.

Do you mean that a single class would be enough to create tests returning objects and allowing reusing these objects in more complex tests, replacing, in part, the use of fixtures? I really have a lot to learn coming from the Rust lang. I'll look for it in the booklets. :)

**Comment by Ducasse** [comment]
Yes.

```
In this section, we show that SUnit offers two hooks to define what a test
selector is and how to perform the test.
      HiExamplesTest class >> classWithExamplesToTest
        ^ self
      HiExamplesTest class >> isAbstract
        ^ self = HiExamplesTest
      HiExamplesTest class >> testSelectors [
        ^ self classWithExamplesToTest classSide methods
          select: [ :each |  (each selector beginsWith: 'example') and: [
           each numArgs = 0 ] ]
          thenCollect: [ :each | each selector ]
      HiExamplesTest >> performTest
        example := self class classWithExamplesToTest perform:
           testSelector asSymbol
```

**Comment by Ducasse** [comment]
Here we define class methods but it could be instance methods.

**Comment by Ducasse** [comment]
I never understood why GT needed to reinvent the wheel and not just spend a couple of hours to provide a solution. This example is in P9. We could use a pragma if we want.
